### PR TITLE
[python] checking if "RequestExecuter" exist before creating a new in…

### PR DESCRIPTION
**Pull request by Or-Geva**
…stance

### CODE REVIEW

File: pyravendb/store/document_store.py [hunk 1]
```diff
@@ -98,10 +98,11 @@ def get_request_executor(self, db_name=None):
         db_name = db_name if db_name is not None else self.database
         if db_name is None:
             raise ValueError("database name cannot be None")
-
-        with self.lock:
-            self._request_executors.setdefault(db_name, RequestsExecutor.create(self.urls, db_name, self._certificate,
-                                                                                self.conventions))
+        if self._request_executors.get(db_name) is None:
```
Comment (ayende) [inline pyravendb/store/document_store.py]: We are mutating and reading the dic outside the lock.
This _looks_ wrong.

Can you ensure that this won't cause race conditions? 
Can you check inside the lock as well?

**Review by ayende — CHANGES_REQUESTED** [review]

File: pyravendb/store/document_store.py (hunk 1 above)
Comment (Or-Geva) [reply pyravendb/store/document_store.py]: Thanks for the heads-up!!, will add a new test as well.

File: pyravendb/tests/database_changes/test_database_changes.py [hunk 2]
```diff
@@ -67,6 +67,12 @@ def test_for_document(self):
         self.assertEqual('Put', documents[0]["Type"])
         self.assertIsNotNone(documents[0]["ChangeVector"])
 
+    def test_for_max_document(self):
```
Comment (ppekrol) [inline pyravendb/tests/database_changes/test_database_changes.py]: why this test is in Database_Changes tests?

Comment (Or-Geva) [reply pyravendb/tests/database_changes/test_database_changes.py]: Fixed.

File: pyravendb/store/document_store.py [hunk 3]
```diff
@@ -100,7 +100,8 @@ def get_request_executor(self, db_name=None):
             raise ValueError("database name cannot be None")
 
         with self.lock:
-            self._request_executors.setdefault(db_name, RequestsExecutor.create(self.urls, db_name, self._certificate,
+            if self._request_executors.get(db_name) is None:
+                self._request_executors.setdefault(db_name, RequestsExecutor.create(self.urls, db_name, self._certificate,
                                                                                 self.conventions))
         return self._request_executors[db_name]
```
Comment (ayende) [inline pyravendb/store/document_store.py]: You are still reading on one thread, modifying on the other, though.

**Review by ayende — CHANGES_REQUESTED** [review]

File: pyravendb/store/document_store.py (hunk 3 above)
Comment (ppekrol) [reply pyravendb/store/document_store.py]: cc @gregolsky

Comment (Or-Geva) [reply pyravendb/store/document_store.py]: how about including the return statement inside the lock?

**Review by ayende — APPROVED** [review]

**Pull request merged**